app/services/catalog/override_icon.rb

```diff
@@ -0,0 +1,17 @@
+module Catalog
+  class OverrideIcon
+    attr_reader :icon
+
+    def initialize(icon_id, portfolio_item_id)
+      @icon = Icon.find(icon_id)
+      @portfolio_item = PortfolioItem.find(portfolio_item_id)
+    end
+
+    def process
+      @portfolio_item.icon.destroy
```

Comment:
Should this be a `destroy!` 
Also, is this a soft destroy?  Have we had that conversation?

Reply:
No we hadn't but after discussing I implemented using the `Catalog::SoftDelete` service class which I used for portfolios/portfolio_items as well.